rasa/utils/llm.py

```diff
@@ -0,0 +1,104 @@
+from typing import Optional
+import openai
+import openai.error
+import structlog
+from rasa.shared.core.events import BotUttered, UserUttered
+
+from rasa.shared.core.trackers import DialogueStateTracker
+
+structlogger = structlog.get_logger()
+
+USER = "USER"
+
+AI = "AI"
+
+DEFAULT_OPENAI_GENERATE_MODEL_NAME = "text-davinci-003"
+
+DEFAULT_OPENAI_CHAT_MODEL_NAME = "gpt-3.5-turbo"
+
+DEFAULT_OPENAI_TEMPERATURE = 0.7
+
+
+def generate_text_openai_chat(
+    prompt: str,
+    model: str = DEFAULT_OPENAI_CHAT_MODEL_NAME,
+    temperature: float = DEFAULT_OPENAI_TEMPERATURE,
+) -> Optional[str]:
+    """Generates text using the OpenAI chat API.
+
+    Args:
+        prompt: the prompt to send to the API
+        model: the model to use for generation
+        temperature: the temperature to use for generation
+
+    Returns:
+        The generated text.
+    """
+    # TODO: exception handling
+    try:
+        chat_completion = openai.ChatCompletion.create(  # type: ignore[no-untyped-call]
+            model=model,
+            messages=[{"role": "user", "content": prompt}],
+            temperature=temperature,
+        )
+        return chat_completion.choices[0].message.content
+    except openai.error.OpenAIError:
+        structlogger.exception("openai.generate.error", model=model, prompt=prompt)
+        return None
```

Comment:
mhm don't have a strong opinion on this but would say the suggestion is harder to read as it merges the execution paths again. there is also an if missing, so it would be

```   
    generated_text = None

    try:
        chat_completion = openai.ChatCompletion.create(  # type: ignore[no-untyped-call]
            model=model,
            messages=[{"role": "user", "content": prompt}],
            temperature=temperature,
        )
    except openai.error.OpenAIError:
        structlogger.exception("openai.generate.error", model=model, prompt=prompt)

    if chat_completion and chat_completion.choices:
        return chat_completion.choices[0].message.content
    return generated_text
```

Reply:
Don't have a strong opinion either